net-vpn/i2p/i2p-0.9.33.ebuild

```diff
@@ -0,0 +1,159 @@
+
+LICENSE="Apache-2.0 Artistic BSD CC-BY-2.5 CC-BY-3.0 CC-BY-SA-3.0 EPL-1.0 GPL-2 GPL-3 LGPL-2.1 LGPL-3 MIT public-domain WTFPL-2"
+SLOT="0"
+
+# Until the deps reach other arches
+KEYWORDS="~amd64 ~x86"
+IUSE="+ecdsa nls"
+
+# dev-java/ant-core is automatically added due to java-ant-2.eclass
+CP_DEPEND="dev-java/bcprov:1.50
+	dev-java/jrobin:0
+	dev-java/slf4j-api:0
+	dev-java/tomcat-jstl-impl:0
+	dev-java/tomcat-jstl-spec:0
+	dev-java/java-service-wrapper:0"
+
+DEPEND="${CP_DEPEND}
+	dev-java/eclipse-ecj:*
+	dev-libs/gmp:0
+	nls? ( >=sys-devel/gettext-0.19 )
+	>=virtual/jdk-1.7"
+
+RDEPEND="${CP_DEPEND}
+	ecdsa? (
+		|| (
+			dev-java/icedtea:8[-sunec]
+			dev-java/icedtea-bin:8
+			dev-java/oracle-jre-bin
+			dev-java/oracle-jdk-bin
+		)
+	)
+	!ecdsa? ( >=virtual/jre-1.7 )"
+
+EANT_BUILD_TARGET="pkg"
+JAVA_ANT_ENCODING="UTF-8"
+
+I2P_ROOT='/usr/share/i2p'
+I2P_CONFIG_HOME='/var/lib/i2p'
+I2P_CONFIG_DIR="${I2P_CONFIG_HOME}/.i2p"
+
+RES_DIR='installer/resources'
+
+pkg_setup() {
+	java-pkg-2_pkg_setup
+
+	enewgroup i2p
+	enewuser i2p -1 -1 "${I2P_CONFIG_HOME}" i2p
+}
+
+src_unpack() {
+	unpack ${A}
+	cd "${S}" || die
+	java-ant_rewrite-classpath
+}
+
+src_prepare() {
+	java-pkg-2_src_prepare
+
+	# We're on GNU/Linux, we don't need .exe files
+	echo "noExe=true" > override.properties || die
+	if ! use nls; then
+		echo "require.gettext=false" >> override.properties || die
+	fi
+
+	# avoid auto starting browser
+	sed -i 's|clientApp.4.startOnLoad=true|clientApp.4.startOnLoad=false|' \
+		"${RES_DIR}/clients.config" || die
+
+	# generate classpath, keeping the default to be replaced later
+	local lib cp i=2
+	for lib in $(echo "$CP_DEPEND" | cut -d / -f 2 | sed 's|:0$||')
+	do
+		cp+="wrapper.java.classpath.$((i++))=$(java-pkg_getjars $lib)\n"
+	done
+	sed -e "s|\(wrapper\.java\.classpath\.1=.*\)|\1\n$cp|" \
+		-i "${RES_DIR}/wrapper.config" || die
+
+	# wrapper specific fixes
+	sed -e "s|\(wrapper\.java\.command\)=.*|\1=/etc/java-config-2/current-system-vm/bin/java|" \
+		-e "s|\(wrapper\.java\.library\.path\.1\)=.*|\1=/usr/lib/java-service-wrapper|" \
+		-i "${RES_DIR}/wrapper.config" || die
+
+	# we do it now so we can resolve path after
+	default
+
+	# replace paths as the installer would
+	sed -i "s|%INSTALL_PATH|${I2P_ROOT}|" \
+		"${RES_DIR}/"{eepget,i2prouter,runplain.sh}  || die
+	sed -i "s|\$INSTALL_PATH|${I2P_ROOT}|" "${RES_DIR}/wrapper.config" || die
+	sed -i "s|%SYSTEM_java_io_tmpdir|${I2P_CONFIG_DIR}|" \
+		"${RES_DIR}/"{i2prouter,runplain.sh} || die
+	sed -i "s|%USER_HOME|${I2P_CONFIG_HOME}|" "${RES_DIR}/i2prouter" || die
+}
+
+src_install() {
+	# cd into pkg-temp.
+	cd "${S}/pkg-temp" || die
+
+	# This is ugly, but to satisfy all non-system .jar dependencies, jetty
+	# would need to be packaged. It would be too large a task
+	# for an unseasoned developer. This seems to be the most pragmatic solution
+	java-pkg_jarinto "${I2P_ROOT}/lib"
+	local i
+	for i in BOB commons-el commons-logging i2p i2psnark i2ptunnel \
```

Comment:
Is it different than `lib/*.jar`?

Reply:
Yes, because we don't want to install jar already installed system-wide.

Reply:
Wouldn't it be easier to rm the offending jars? Just asking which is easier for you.

Reply:
You're right, it would be easier (and safer), I'll change it.